tools/template_based_query_generation/src/main/java/Utils.java

```diff
@@ -0,0 +1,185 @@
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Utilities class that provides random and IO helper functions.
+ */
+public class Utils {
+
+	private static final ThreadLocalRandom random = ThreadLocalRandom.current();
+
+	private static final int lowerBound = 0;
+
+	private static final String CHARSET = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_";
+
+	/**
+	 * Returns a random integer between a lowerBound and an upperBound, inclusive
+	 *
+	 * @param upperBound a non-negative integer upper bound on the generated random integer, inclusive
+	 * @return a random integer between lowerBound and upperBound, inclusive
+	 * @throws IllegalArgumentException if upperBound is negative
+	 */
+	public static int getRandomInteger(int upperBound) throws IllegalArgumentException {
+		if (upperBound < 0) {
+			throw new IllegalArgumentException("Upper bound cannot be negative");
+		}
+
+		return random.nextInt(lowerBound, upperBound + 1);
+	}
+
+	/**
+	 * Returns a random string with a specified length that matches the regex '[a-zA-Z_]'
+	 *
+	 * @param length a nonzero integer specifying the desired length of the generated string
+	 * @return a random string that matches the regex '[a-zA-Z_]' and has the specified length
+	 */
+	public static String getRandomString(int length) throws IllegalArgumentException {
+		if (length <= 0) {
+			throw new IllegalArgumentException("Random string must have positive length");
+		}
+
+		StringBuilder sb = new StringBuilder();
+
+		for (int i = 0; i < length; i++) {
+			int randomIndex = (int) (random.nextDouble() * CHARSET.length());
+			char randomChar = CHARSET.charAt(randomIndex);
+			if (i == 0 && Character.isDigit(randomChar)) {
+				// SQL identifiers can't start with digits, so replace with an arbitrary character
+				randomChar = 'a';
```

Comment:
consider use CHARSET.charAt(random.nextInt(0,52)) ?

Reply:
Good catch, will change it.